sklearn/tree/_criterion.pxd

```diff
@@ -40,6 +40,14 @@ cdef class Criterion:
     cdef double weighted_n_left          # Weighted number of samples in the left node
     cdef double weighted_n_right         # Weighted number of samples in the right node
 
+    cdef double* node_sum_total          # An array of the current node value
+    cdef double* node_sum_left           # An array of the left node value
+    cdef double* node_sum_right          # An array of the right node value
+
+    cdef double sq_sum_total          # Cumulative square in the current node
+    cdef double sq_sum_left           # Cumulative square in the left node
+    cdef double sq_sum_right          # Cumulative square in the right node                
```

Comment:
If this are used only in regression, they should be declared in RegressionCriterion.